tests/python/unittest/test_tir_schedule_transform_layout.py

```diff
@@ -0,0 +1,160 @@
+    for i, j in T.grid(128, 128):
+        with T.block("C"):
+            vi, vj = T.axis.remap("SS", [i, j])
+            C[vi // 16, vj // 16, vi % 16, vj % 16] = B[vi, vj] + 1.0
+
+
+@T.prim_func
+def permuted_shared_memory(a: T.handle, c: T.handle) -> None:
+    A = T.match_buffer(a, (128, 128))
+    C = T.match_buffer(c, (128, 128))
+    A_shared = T.alloc_buffer((128, 128), scope="shared")
+    for i0, j0, in T.grid(32, 4):
+        for fused_i1_j1 in T.thread_binding(0, 32, 'threadIdx.x'):
+            for j2 in T.vectorized(0, 4):
+                with T.block("A_shared"):
+                    vi = T.axis.S(128, i0 * 4 + fused_i1_j1 // 8)
+                    vj = T.axis.S(128, j0 * 32 + fused_i1_j1 % 8 * 4 + j2)
+                    A_shared[vi, vj] = A[vi, vj]
+    for i, j in T.grid(128, 128):
+        with T.block("C"):
+            vi, vj = T.axis.remap("SS", [i, j])
+            C[vi, vj] = A_shared[vi, vj] + 1.0
+
+
+@T.prim_func
+def permuted_shared_memory_transformed(a: T.handle, c: T.handle) -> None:
+    A = T.match_buffer(a, (128, 128))
+    C = T.match_buffer(c, (128, 128))
+    A_shared = T.alloc_buffer((32, 4, 4, 32), scope="shared")
+    for i0, j0, in T.grid(32, 4):
+        for fused_i1_j1 in T.thread_binding(0, 32, 'threadIdx.x'):
+            for j2 in T.vectorized(0, 4):
+                with T.block("A_shared"):
+                    vi = T.axis.S(128, i0 * 4 + fused_i1_j1 // 8)
+                    vj = T.axis.S(128, j0 * 32 + fused_i1_j1 % 8 * 4 + j2)
+                    A_shared[vi // 4, vj // 32, vi % 4, (((vj % 32) // 8) ^ (vi % 4)) + vj % 8] = A[vi, vj]
+    for i, j in T.grid(128, 128):
+        with T.block("C"):
+            vi, vj = T.axis.remap("SS", [i, j])
+            C[vi, vj] = A_shared[vi // 4, vj // 32, vi % 4, (((vj % 32) // 8) ^ (vi % 4)) + vj % 8] + 1.0
+
+
+# pylint: enable=no-member,invalid-name,unused-variable,line-too-long,redefined-outer-name,unexpected-keyword-arg,too-many-nested-blocks
+# fmt: on
+
+
+def test_two_elementwise_transform_intermediate_buffer():
+    sch = tir.Schedule(two_elementwise, debug_mask="all")
+    block = sch.get_block("B")
+    sch.transform_layout(block, 0, False, packed_index_map_func)
+    tvm.ir.assert_structural_equal(two_elementwise_transformed_intermediate_buffer, sch.mod["main"])
+    verify_trace_roundtrip(sch=sch, mod=two_elementwise)
+
+
+def test_two_elementwise_transform_input_buffer():
+    sch = tir.Schedule(two_elementwise, debug_mask="all")
+    block = sch.get_block("B")
+    sch.transform_layout(block, 0, True, packed_index_map_func)
+    print(sch.mod["main"].script())
+    tvm.ir.assert_structural_equal(two_elementwise_transformed_input_buffer, sch.mod["main"])
+    verify_trace_roundtrip(sch=sch, mod=two_elementwise)
+
+
+def test_two_elementwise_transform_output_buffer():
+    sch = tir.Schedule(two_elementwise, debug_mask="all")
+    block = sch.get_block("C")
+    sch.transform_layout(block, 0, False, packed_index_map_func)
```

Comment:
Can we use the lambda function here? If yes, please add (or update one) test here.